src/MongoDB.Driver.Core/Core/Servers/Server.cs

```diff
@@ -279,21 +279,21 @@ private void HandleChannelException(IConnection connection, Exception ex)
             if (ShouldInvalidateServer(ex))
             {
                 var shouldClearConnectionPool = ShouldClearConnectionPoolForChannelException(ex, connection.Description.ServerVersion);
-                Invalidate(shouldClearConnectionPool);
+                Invalidate($"ChannelException:{ex.GetType().Name}", shouldClearConnectionPool);
```

Comment:
In general, I would suggest adding an enum instead of just strings. But it's obviously not the case if the value is depended on exception type/message

Reply:
We could consider adding the full message... but I think that might make for a string that is excessively long.

Please comment again indicating whether you think we should make the reason string longer to include the message. I'm inclined against but could be convinced.

Reply:
I think adding only a type name here will be helpful but not in all cases. What if the important exception will be inside the innerException? Or if we need to know the error code in the current exception? Of course, all this information will be available in the stacktrace, but in this case, maybe it's enough to add only `ChannelException` here?

I think adding the whole description of the message will be the most helpful and useful. The fact that the same error information will be presented in the stacktrace doesn't look too bad to me (especially taking the fact that this case will be possible only for exceptions into account).